packages/react-ui-core/src/Gmap/FreeDrawLayer.js

```diff
@@ -0,0 +1,199 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import autobind from 'autobind-decorator'
+import themed from 'react-themed'
+import { EVENTS } from './utils/const'
+import { setupEvents, removeEvent } from './utils/mapEventHelpers'
+
+@themed(['Gmap_FreeDrawTootip'], { pure: true })
+export default class FreeDrawLayer extends PureComponent {
+  static propTypes = {
+    map: PropTypes.object,
+    onMapDrawStart: PropTypes.func,
+    onMapDrawEnd: PropTypes.func,
+    theme: PropTypes.object,
+    mapControls: PropTypes.object,
+    shapes: PropTypes.object,
+    multipleShapes: PropTypes.bool,
+    disabled: PropTypes.bool,
+    dataStyle: PropTypes.object,
+  }
+
+  static defaultProps = {
+    theme: {},
+    map: {},
+    multipleShapes: false,
+    disabled: false,
+    shapes: {},
+    dataStyle: {},
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      drawing: false,
+    }
+    this.polygons = (Object.keys(this.props.shapes).length > 0) ? [this.addShapeToGMaps(
```

Comment:
I think you can just do `this.polygons = Object.keys(this.props.shapes).length ? [this.addShapeToGMaps(`

Reply:
Anyway possible to clean this up a little? It's a bit difficult to follow.